08_multibundle_requirejs/Gruntfile.js

```diff
@@ -0,0 +1,286 @@
+var path = require('path'),
+    async = require('async'),
+    watchFilesLess = 'assets/style/**/*.less',
+
+    Mapper = require('multibundle-mapper'),
+    usersBundleTemplatePath = 'app/templates/users_bundle/**/*.hbs',
+    reposBundleTemplatePath = 'app/templates/repos_bundle/**/*.hbs';
+
+
+module.exports = function(grunt) {
+
+    // Project configuration.
+    grunt.initConfig({
+        pkg: grunt.file.readJSON('package.json'),
+
+        // Configuration to be run (and then tested).
+        'multibundle-requirejs': {
+            options: {
+                '_config': {
+                    // 4 is silent in r.js world
+                    logLevel: process.env.quiet ? 4 : 1,
+                    destination: 'public/js',
+                    sharedBundles: ['common'],
+                    // or custom function `hashFiles(output, componentOptions)`
+                    hashFiles: true,
+                    handleMapping: function(options, grunt, done) {
+                        // For this example, write to the local config.
+                        // For production, prefix should be configured by environment to build for CDNs, etc.
+                        return Mapper.json('config/local.json', {
+                                prefix: '/js'
+                            },
+                            function(err) {
+                                console.log('Mapping meta data has been written.', err);
+                                // grunt style callback, so that we can 
+                                done(err ? false : true);
+                            });
+                    },
+
+                    // pass options to r.js
+                    baseUrl: '.',
+                    optimize: 'uglify',
+                    // optimize: 'none',
+                    sharedPaths: {
+                        // test location namespacing
+                        'app': 'app',
+                        'assets': 'assets',
+                        // needed for rendr modules
+                        'rendr': 'node_modules/rendr'
+                    },
+                    preserveLicenseComments: false
+                },
+
+                // Creates `<destination>/common.<hash>.js` file that includes all the modules specified in the bundle,
+                // shared modules between all the pages.
```

Comment:
I think for example apps single Gruntfile is more descriptive and more familiar to novice eye.
When real project grow eve can find their way to make it more manageable. Like ditching grunt and switching to npm scripts :)